packages/strapi-plugin-i18n/constants/index.js

```diff
@@ -2,12 +2,35 @@
 
 const isoLocales = require('./iso-locales');
 
-const DEFAULT_LOCALE = {
-  name: 'English',
-  code: 'en',
+/**
+ * Returns the default locale based either on env var or english
+ * @returns {string}
+ */
+const getInitLocale = () => {
+  const envLocaleCode = process.env.STRAPI_PLUGIN_I18N_INIT_LOCALE_CODE;
+
+  if (envLocaleCode) {
+    const matchingLocale = isoLocales.find(({ code }) => code === envLocaleCode);
+
+    if (!matchingLocale) {
+      throw new Error(
+        'Unknown locale code provided in the environment variable STRAPI_PLUGIN_I18N_INIT_LOCALE_CODE'
+      );
+    }
+
+    return { ...matchingLocale };
```

Comment:
Why do we need to create a new version of the matchingLocale here?

Reply:
to avoid any mutation to the value in isoLocales